interfaces/utils/path_patterns_test.go

```diff
@@ -0,0 +1,130 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package utils_test
+
+import (
+	"regexp"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces/utils"
+)
+
+type pathPatternsSuite struct{}
+
+var _ = Suite(&pathPatternsSuite{})
+
+func (s *pathPatternsSuite) TestRegexCreationHappy(c *C) {
+	// to save some typing:
+	d := utils.GlobDefault
+	n := utils.GlobNull
+
+	data := []struct {
+		pattern       string
+		glob          utils.GlobFlags
+		expectedRegex string
+	}{
+		{`/media/user/`, d, `^/media/user/$`},
+		{`/dev/sd*`, d, `^/dev/sd[^/\x00]*$`},
+		{`/dev/sd*`, n, `^/dev/sd[^/]*$`},
+		{`/dev/sd?`, d, `^/dev/sd[^/\x00]$`},
+		{`/etc/**`, d, `^/etc/[^/\x00][^\x00]*$`},
+		{`/home/*/.bashrc`, d, `^/home/[^/\x00][^/\x00]*/\.bashrc$`},
+		{`/home/*/.bashrc`, n, `^/home/[^/][^/]*/\.bashrc$`},
+		{`/media/{user,loser}/`, d, `^/media/(user|loser)/$`},
+		{`/nested/{a,b{c,d}}/`, d, `^/nested/(a|b(c|d))/$`},
+		{`/media/\{in-braces\}/`, d, `^/media/\{in-braces\}/$`},
+		{`/media/\[in-brackets\]/`, d, `^/media/\[in-brackets\]/$`},
+		{`/dev/sd[abc][0-9]`, d, `^/dev/sd[abc][0-9]$`},
+		{`/quoted/bracket/[ab\]c]`, d, `^/quoted/bracket/[ab\]c]$`},
+		{`{[,],}`, d, `^([,]|)$`},
+		{`/path/with/comma[,]`, d, `^/path/with/comma[,]$`},
+	}
+
+	for _, testData := range data {
+		pattern := testData.pattern
+		expectedRegex := testData.expectedRegex
+		regex, err := utils.CreateRegex(pattern, testData.glob)
+		c.Assert(err, IsNil, Commentf("%s", pattern))
+		c.Assert(regex, Equals, expectedRegex, Commentf("%s", pattern))
+		// Also, make sure that the obtained regex is valid
+		_, err = regexp.Compile(regex)
+		c.Assert(err, IsNil, Commentf("%s", pattern))
+	}
+}
+
+func (s *pathPatternsSuite) TestRegexCreationUnhappy(c *C) {
+	data := []struct {
+		pattern       string
+		expectedError string
+	}{
+		{`/media/{}/`, `Invalid number of items between {}:.*`},
+		{`/media/{some/things`, `Missing 1 closing brace\(s\):.*`},
+		{`/media/}`, `Invalid closing brace, no matching open { found:.*`},
+		{`/media/[abc`, `Missing closing bracket ']':.*`},
```

Comment:
what about a test case for `{a,,b}` where we have an extra `comma` ? does that work or should we fail on that?

Reply:
It works, it means that you can have `a`, `b`, or nothing.